cmake/FindMOAB.cmake

```diff
@@ -10,15 +10,28 @@ find_path(MOAB_CMAKE_CONFIG
   PATHS ${MOAB_SEARCH_DIRS}
   NO_DEFAULT_PATH
 )
-if (MOAB_CMAKE_CONFIG)
+
+# First check if we are forcing the download of MOAB
+if (PULL_INSTALL_MOAB)
```

Comment:
Have we thought about whether we put this condition up one level?  That is, if we are going to PULL_INSTALL, then we don't call `FindPackage` at all and just do this macro instead?  Then we don't need to mess with the `FindMOAB` file.  I think that will work???